### PR TITLE
Refactor Scan Page

### PR DESCRIPTION
**Pull request by XanderLuciano**
Refactors Scan Page into more discrete components. The components on the page are still fairly tightly coupled, but separating into separate component files makes the separation of functions clearer and improves maintainability of each individual aspect.

Along with the refactor, I also included a few improvements of existing functions. I have changed some synchronous functions into Async/Await promises that are no longer blocking. I've also added some new UX features to communicate to the user what the current state is for the camera (stopped vs. loading vs. scanning).

This also improves the UX of the scanning with camera feature by preventing the dom element from resizing when the camera starts/stops.

Barcodes are now automatically searched upon successful scan. 

I have more planned for this page, but felt the current state of this PR is at a good state to push upstream while I work on new features.

Example Functionality:
ScanPage-1

### CODE REVIEW

**Comment by XanderLuciano** [comment]
tracks towards: #5748

File: src/frontend/src/pages/scan/BarcodeInputManual.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+// Region input stuff
+enum InputMethod {
+  Manual = 'manually',
+  ImageBarcode = 'imageBarcode'
+}
+
+interface inputProps {
+  action: (items: ScanItem) => void;
+}
```
Comment (matmair) [inline src/frontend/src/pages/scan/BarcodeInputManual.tsx:30]: Blocking: This is defined in multiple files, there should be one shared interface

**Review by matmair — CHANGES_REQUESTED** [review]
I see multiple problems in a quick test:
1) Constant errors that no QR code was found making notifications unusable
2) Unknown codes are just disregarded - and do not show up in history; this is a blocking bug

**Comment by XanderLuciano** [comment]
> Constant errors that no QR code was found

Hmm, I encountered this at one point and thought the string comparison would suffice, but probably need to revert to just dump the errors to console instead of the pop-up notification. I was hoping to pass meaningful errors up to the user, but if it results in an error notification on every frame, that's bad.

> Unknown codes are just disregarded - and do not show up in history

What is the intended behavior for scanning invalid QR codes? Just showing the raw QR code string value in the history?

Does showing invalid data in the history provide meaningful information to the user? Or would a popup notification with an error message and the QR code string also provide sufficient feedback to the user?

File: src/frontend/src/pages/scan/BarcodeInputManual.tsx (hunk 1 above)
Comment (XanderLuciano) [reply src/frontend/src/pages/scan/BarcodeInputManual.tsx]: Moved into common interface file for reuse

**Pull request closed without merging**